templates/test/ci/cluster-template-prow.yaml

```diff
@@ -2,6 +2,7 @@ apiVersion: cluster.x-k8s.io/v1beta1
 kind: Cluster
 metadata:
   labels:
+    cloud-provider: ${CLOUD_PROVIDER_AZURE_LABEL:=azure}
     cni: calico
```

Comment:
Q: Is this still needed? Seems like you renamed the label patch from `cni: calico` to `cloud-provider: azure-flatcar` but maybe the label is coming from somewhere else. (non-blocking)

Reply:
> Seems like you renamed the label patch from cni: calico to cloud-provider: azure-flatcar

Where do you see this? only the flatcar template is using `cloud-provider: azure-flatcar`, the others use `cloud-provider: azure` or `cloud-provider: azure-ci`

Reply:
I meant in reference to [this file](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/cede1c5597162fa215c75e3afea9959b761375bc/templates/test/ci/patches/cluster-label-cloud-provider-azure-flatcar.yaml). It used to be named cluster-label-calico.yaml, was that old file referenced in any of the other kustomization.yamls?

Reply:
This just looks like some funky renamed file accounting on git's part to me. @Jont828 do things look okay to you just looking at the plain file tree? https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/cede1c5597162fa215c75e3afea9959b761375bc/templates/test/ci/patches/cluster-label-calico.yaml

Reply:
Yep, the file seems to be there!

Reply:
Yes I think that's just diff weirdness, final result lgtm